tests/test_task_worker.py

```diff
@@ -672,6 +672,36 @@ async def register_localfilesystem(self):
         """Register LocalFileSystem before running tests to avoid race conditions."""
         await LocalFileSystem.register_type_and_schema()
 
+    async def test_task_worker_limiter_gracefully_handles_same_task_run(
+        self, prefect_client
+    ):
+        @task
+        def slow_task():
+            import time
+
+            time.sleep(1)
+
+        task_worker = TaskWorker(slow_task, limit=1)
+
+        task_run_future = slow_task.apply_async()
+        task_run = await prefect_client.read_task_run(task_run_future.task_run_id)
+
+        try:
+            with anyio.move_on_after(1):
+                # run same task, one should acquire a token
+                # the other will gracefully be skipped.
+                async with task_worker:
+                    await asyncio.gather(
+                        task_worker.execute_task_run(task_run),
+                        task_worker.execute_task_run(task_run),
```

Comment:
On `main` this errors with `this borrower is already holding one of this CapacityLimiter's tokens`